probe-rs/src/architecture/arm/core/exception_handling/armv6m_armv7m_shared.rs

```diff
@@ -66,20 +201,35 @@ pub(crate) fn exception_details(
             )
         })?
         .try_into()?;
+    Ok(frame_return_address)
+}
 
-    if ExcReturn(frame_return_address).is_exception_flag() == 0xF {
-        // This is an exception frame.
-
-        Ok(Some(ExceptionInfo {
-            description: exception_interface
-                .exception_description(memory_interface, stackframe_registers)?,
-            calling_frame_registers: exception_interface
-                .calling_frame_registers(memory_interface, stackframe_registers)?,
-        }))
-    } else {
-        // This is a normal function return.
-        Ok(None)
+pub(crate) fn raw_exception(
+    stackframe_registers: &crate::debug::DebugRegisters,
+) -> Result<u32, Error> {
+    // Load the provided xPSR register as a bitfield.
+    let mut exception_number = Xpsr(
+        stackframe_registers
+            .get_register_value_by_role(&crate::core::RegisterRole::ProcessorStatus)?
+            as u32,
+    )
+    .exception_number();
+    if exception_number.is_zero()
+        && stackframe_registers
+            .get_register_value_by_role(&crate::core::RegisterRole::ReturnAddress)?
+            == 0xFFFF_FFFF
+    {
+        // Although the exception number is 0, for the purposes of unwind, this treated as a reset exception.
+        // Based on the sections, "The special-purpose program status registers, xPSR"
+        // and "Reset Behaviour" in the ARMv7-m Architecture Reference Manual,
+        // - "On reset, the processor is in Thread mode and ...
+        //   - ... the Exception Number field of the IPSR is cleared to 0. As a result, the value 1, the exception number for reset,
+        //    is a transitory value, that software cannot see as a valid IPSR Exception Number."
+        //   - The LR register value is set to 0xFFFFFFFF (The reset value)
+        exception_number = 1;
```

Comment:
I am not sure I understand why we manipulate this :/

Reply:
In the target registers, it is a transitory value, that gets reset to 0 by the time we get here. We are not manipulating the register value, just recording the exception number in the StackFrame we are reporting.

Reply:
Ah, so we are basically too slow to retrieve the value of 1?

Reply:
All debuggers would be too slow. It resets the value to zero before it resets the LR register (which is the register debuggers use as EXC_RETURN)

Reply:
Ah yeah that's what I meant :) It was my understanding that it cannot be done differently. Seems a bit odd to me to be fair.
Maybe you can reword it a bit in the comment so other dimwits like me will understand as well :D

Reply:
Done ... hopefully that helps :)